File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/swift/F08.P2.swift
func probe(_ a: Int32, _ b: Int32) -> Int32 {
// BEGIN PROBE F08.P2
    let d = a.dividedReportingOverflow(by: b)
    let q = d.overflow ? 0 : d.partialValue
    return q
// END PROBE F08.P2
}

print(probe(7, 0))

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F08.P2.go
package main

import "fmt"

func probe(a, b int32) int32 {
	// BEGIN PROBE F08.P2
	var q int32
	if b != 0 {
		q = a / b
	}
	return q
	// END PROBE F08.P2
}

func main() {
	fmt.Println(probe(7, 0), probe(7, 2))
}
